client/theia-ecore/src/node/ecore-file-generation.ts

```diff
@@ -58,18 +59,90 @@ export class EcoreFileGenServer implements FileGenServer, BackendApplicationCont
             process.process.on("exit", (code: any) => {
                 switch (code) {
                     case 0: resolve("OK"); break;
-                    case -10: resolve("Name Parameter missing"); break;
-                    case -11: resolve("NsPrefix Parameter missing"); break;
-                    case -12: resolve("NsURI Parameter missing"); break;
-                    case -13: resolve("Workspace Path Parameter missing"); break;
-                    case -20: resolve("Encoding not found, check Server Log!"); break;
-                    case -30: resolve("IO Exception occurred, check Server Log!"); break;
                     default: resolve("UNKNOWN ERROR"); break;
                 }
             });
         });
     }
 
+    generateGenModel(workspacePath: string,ecorePath: string, customPackageName: string, folderName: string): Promise<string> {
+        const jarPath = path.resolve(__dirname, "..", "..", "..", "..",
+            "server", "org.eclipse.emfcloud.ecore.backend-app", "org.eclipse.emfcloud.ecore.codegen.product",
+            "target", "products", "org.eclipse.emfcloud.ecore.codegen.product", "linux", "gtk", "x86_64", "plugins",
+            "org.eclipse.equinox.launcher_1.5.600.v20191014-2022.jar");
+        if (jarPath.length === 0) {
+            throw new Error("The eclipse.equinox.launcher is not found. ");
+        }        
+
+        const command = "java";
+        const args: string[] = [];
+        args.push(
+            "-cp", jarPath,
+            "org.eclipse.equinox.launcher.Main",
+            "-data", workspacePath,
+            "-application", "org.eclipse.emfcloud.ecore.backend.app.create-genmodel",
+            ecorePath, customPackageName, folderName
+        );
+        
+        return new Promise(resolve => {
+            const process = this.spawnProcess(command, args);
+            // eslint-disable-next-line no-null/no-null
+            if (process === undefined || process.process === undefined || process === null || process.process === null) {
+                resolve("Process not spawned");
+                return;
+            } else {
+                if (process.process.stdout){
+                    process.process.stdout.on("data", function (data) {
+                        console.log("stdout: " + data.toString());
+                    });
+                }
+            }
+
+            process.process.on("exit", (code: any) => {
+                switch (code) {
+                    case 0: resolve("OK"); break;
+                    default: resolve("UNKNOWN ERROR " + code); break;
+                }
+            });
+        });
+    }
+
+    generateCode(genmodelPath: string, workspacePath: string): Promise<string> {
+        const jarPath = path.resolve(__dirname, "..", "..", "..", "..",
```

Comment:
We should probably have a single method for this. We should also be able to call directly the `ecore_glsp_backend` binary, which would simplify the command line a lot. Something like this:

`ecore_glsp_backend -nosplash [app-args]`

Rather than:

`java -jar -cp <equinox launcher> <Main> [app-args]`